k8s/upgrade/src/bin/upgrade-job/helm/upgrade.rs

```diff
@@ -228,8 +277,18 @@ impl HelmUpgradeBuilder {
             core_chart_extra_args,
             from_version,
             to_version,
+            // Defaults to true, if the value is absent from the helm chart values yaml.
+            ha_is_enabled,
             upgrade_values_file,
-        })
+        };
+
+        // Post-build jobs
+        let read_only_obj = Rc::new(built_obj);
+        for post_build_job in self.post_build_jobs {
+            post_build_job.deref()(Rc::clone(&read_only_obj));
+        }
+
+        Rc::try_unwrap(read_only_obj).map_err(|_| UnwrapRcHelmUpgrade.build())
```

Comment:
This is the last line of the build() method for HelmUpgradeBuilder. Ideally it'd return the built_obj. But, we gather a lot of information while building. Today, we need a way to export the state of HA from this function. So what I have done is added a `post_build_jobs` builder option, which are closures that one could run post-build, to do things. In this change, I've added a closure to update the drain decision.

In file k8s/upgrade/src/bin/upgrade-job/upgrade.rs, line #<span></span>51, I pass in code to read data from the built HelmUpgrade.

Reply:
I've updated the PR with a new commit. I must has overthought about what I am trying to do there, seems straightforward now that I look at it. Anyhow, I've made the changes with a new commit, just in case we want to go back to the older way for some reason.